pkg/skaffold/docker/remote.go

```diff
@@ -62,21 +46,41 @@ func RemoteDigest(identifier string) (string, error) {
 }
 
 // RetrieveRemoteConfig retrieves the remote config file for an image
-func RetrieveRemoteConfig(identifier string) (*v1.ConfigFile, error) {
-	img, err := remoteImage(identifier)
+func RetrieveRemoteConfig(identifier string, insecureRegistries map[string]bool) (*v1.ConfigFile, error) {
+	img, err := remoteImage(identifier, insecureRegistries)
 	if err != nil {
 		return nil, errors.Wrap(err, "getting image")
 	}
 
 	return img.ConfigFile()
 }
 
-func remoteImage(identifier string) (v1.Image, error) {
-	ref, err := name.ParseReference(identifier, name.WeakValidation)
+func remoteImage(identifier string, insecureRegistries map[string]bool) (v1.Image, error) {
+	ref, err := name.ParseReference(identifier)
 	if err != nil {
 		return nil, errors.Wrap(err, "parsing initial ref")
 	}
 
+	if isInsecure(ref.Context().Registry.Name(), insecureRegistries) {
+		ref, err = getInsecureRegistryImpl(identifier)
+		if err != nil {
+			logrus.Warnf("error getting insecure registry: %s\nremote references may not be retrieved", err.Error())
```

Comment:
Sorry, i am not aware of how the go container registry works.
In case we see an error when parsing an insecure registry, we still try to fetch the image? Shd we not fail instead of warning?
Or is there a case where you can still retrieve the insecure image.
#failFast.

Reply:
mmm yeah it's unlikely that retrieving the image will work if this fails, but I don't see the harm in trying. `remote.Image` should be a pretty inexpensive call, and if that fails skaffold will fail.